### ISSUE TITLE
Unclosed markup error if span text goes past line

### ISSUE THREAD

**Issue by mlos100** [ChordPro/chordpro#355]
## Describe the bug
When the text associated with a span element goes beyond the space allocated for one line, an Unclosed markup error occurs in the ChordPro Warning dialog.

The error does not occur if the lyrics plus Pango markup fits on the same line.

```
 : Markup error: " <span font_desc="Helvetica italic 9" foreground="#666666">(to Chorus then"
  Unclosed markup: <span font_desc="helvetica italic 9" foreground="#666666">
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: " <span font_desc="Helvetica italic 9" foreground="#666666">(to Chorus"
  Unclosed markup: <span font_desc="helvetica italic 9" foreground="#666666">
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: " <span font_desc="Helvetica italic 9" foreground="#666666">(to"
  Unclosed markup: <span font_desc="helvetica italic 9" foreground="#666666">
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: " <span font_desc="Helvetica italic 9" foreground="#666666">(to"
  Unclosed markup: <span font_desc="helvetica italic 9" foreground="#666666">
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: "Chorus then Outro)</span>"
  Closing <span> but no markup is open
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: "Chorus then Outro)</span>"
  Closing <span> but no markup is open
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF/Writer.pm line 225.
 : Markup error: "Lift up your song 'Cause you've got a lion in side of those lungs  Get up and praise the  Lord     <span font_desc="Helvetica italic 9" foreground="#666666">(to"
  Unclosed markup: <span font_desc="Helvetica italic 9" foreground="#666666">
 : Markup error: "Chorus then Outro)</span>"
  Closing <span> but no markup is open
 at C:/Program Files/ChordPro.ORG/ChordPro/script/../lib/ChordPro/Output/PDF.pm line 2256.
 : Markup error: "Chorus then Outro)</span>"
  Closing <span> but no markup is open
 : Problems found!

```
### To Reproduce
See Attached chord pro file, span-example.txt

### Expected behavior
Ideally, I think the markup would extend onto the next line.  That said, if the markup didn't fit on the same line, I would end up adjusting either the markup or lyrics so that the note did fit on the same line so I would be fine with an expected error that said the markup didn't fit on the line and was truncated.

## System information
- OS Windows 10
 - Where did you get ChordPro from? GitHub
 - PDF Reader: PDF-Xchange Editor (free)
 - 
ChordPro Preview Editor version 6.050
https://www.chordpro.org
Copyright 2016,2024 Johan Vromans <jvromans@squirrel.nl>

GUI designed with wxGlade

Run-time information:
  ChordPro core          6.050     
  Perl                   v5.30.2    (C:/Program Files/ChordPro.ORG/ChordPro/wxchordpro.exe)
  PPL Packager           1.00      
  CHORDPRO_LIB           ~/lib/ChordPro
  Resource path          ~/lib/ChordPro
                         C:/Program Files/ChordPro.ORG/ChordPro/lib/ChordPro/res
  ABC support            QuickJS_XS (ABC2SVG version v1.22.12 of 2023-12-22)

Modules and libraries:
  wxPerl                 0.9932    
  wxWidgets              3.000002  
  Storable               3.25      
  Object::Pad            0.79      
  Text::Layout           0.032     
  File::LoadLines        1.045     
  PDF::API2              2.043     
  SVGPDF                 0.083     
  Font::TTF              1.06      
  JavaScript::QuickJS    0.18

**Comment by DrCostas** [comment]
Just mentioning this is present in 6.060 and with <i> or <b> or even when the text is inside a {comment: <span>...</span>}

**Comment by sciurius** [comment]
Yes ☹. It is high on the TODO list.

**Comment by gwyndaf** [comment]
As an interim 'solution', would a `linewrap=false` configuration setting be useful (and relatively simple to implement)? That is, to inhibit ChordPro splitting long lines.

If long lines don't get split, that might avoid the markup issue.

Of course, some lines might then run longer than the page width, but a 'nowrap' option puts that in users' hands, and might help avoid the markup errors.

In principle, this could be selectively applied to songlines or comments, but that might add unnecessary complexity for little significant benefit. Maybe simpler as a universal setting.

(Partly raising this in relation to #344, because I often rely on un-wrapped 'over-length' lines to squeeze more into the available page width).

**Comment by sciurius** [comment]
Indeed. One can consider the miscellaneous outputs all to look similar. Should the need rise we can always add a PDF specific setting that overrules the generic default.
